### PR TITLE
Eng 1986 warn user when there is a python version mismatch between server and client

### PR DESCRIPTION
**Pull request by eunice-chan**
## Describe your changes and why you are making these changes
Warn user when there is a python version mismatch between server and client as it may result in unexpected errors when running the workflow on the server.

## Related issue number (if any)
ENG 1986

## Loom demo (if any)
Commented out most of the warning display condition to show what is the warning. I don't have a Python version mismatch so it won't show otherwise.
image

Displays warning there is a mismatch but still allows the workflow to be published (as seen in the next line, there is a URL).

## Checklist before requesting a review
- [x] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [x] I have performed a self-review of my code.
- [x] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [N/A] If this is a new feature, I have added unit tests and integration tests.
- [ ] I have run the integration tests locally and they are passing.
- [x] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [N/A] All features on the UI continue to work correctly.
- [x] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

File: sdk/aqueduct/models/response_models.py [hunk 1]
```diff
@@ -69,9 +69,12 @@ class RegisterWorkflowResponse(BaseModel):
     Attributes:
         id:
             The uuid if of the newly registered workflow.
+        python_version:
+            The Python version in the engine in string format "major_version.minor_version.patch_level".
```
Comment (kenxu95) [inline sdk/aqueduct/models/response_models.py:73]: Can you just provide an example here instead? Like `3.8.11`.

File: src/golang/cmd/server/handler/register_workflow.go [hunk 2]
```diff
@@ -355,5 +359,21 @@ func (h *RegisterWorkflowHandler) Perform(ctx context.Context, interfaceArgs int
 		}
 	}()
 
-	return registerWorkflowResponse{Id: workflowId}, http.StatusOK, nil
+	var version strings.Builder
```
Comment (kenxu95) [inline src/golang/cmd/server/handler/register_workflow.go:362]: Can you note up here that the python version check is best-effort?

File: sdk/aqueduct/client.py [hunk 3]
```diff
@@ -662,7 +663,20 @@ def publish_flow(
         else:
             if run_now is None:
                 run_now = True
-            flow_id = globals.__GLOBAL_API_CLIENT__.register_workflow(dag, run_now).id
+            register_metadata = globals.__GLOBAL_API_CLIENT__.register_workflow(dag, run_now)
```
Comment (kenxu95) [inline sdk/aqueduct/client.py:666]: nit: name `registered_metadata`?

**Review by kenxu95 — APPROVED** [review]
I noticed the error message in your screenshot has the same version for client and server? We only want to print this warning when they mismatch right - was this just a demo oversight?

**Comment by eunice-chan** [comment]
@kenxu95 Good eye! For purposes of demonstration, I removed the condition checking the values are different. When the condition is put back in place, the warning does not show up if there is no mismatch, as expected.

**Pull request merged**